sql/server/src/main/java/io/stargate/api/sql/server/StargateStatement.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.api.sql.server;
+
+import io.stargate.api.sql.plan.PreparedSqlQuery;
+import io.stargate.api.sql.plan.QueryPlanner;
+import io.stargate.db.datastore.DataStore;
+import java.lang.reflect.Type;
+import java.sql.ResultSetMetaData;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.calcite.adapter.java.JavaTypeFactory;
+import org.apache.calcite.avatica.AvaticaParameter;
+import org.apache.calcite.avatica.ColumnMetaData;
+import org.apache.calcite.avatica.Meta;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeField;
+
+public class StargateStatement extends Meta.StatementHandle {
```

Comment:
Here again, since stargate is full of stuff called "statement", I think it's worth finding a name more useful. In general, I'd argue that putting "stargate" into any naming is rarely gonna provide useful information.

Reply:
renamed to `StatementHolder`